jenkins-slave

```diff
@@ -29,6 +29,22 @@
 # * JENKINS_SECRET : agent secret, if not set as an argument
 # * JENKINS_AGENT_NAME : agent name, if not set as an argument
 # * JENKINS_AGENT_WORKDIR : agent work directory, if not set by optional parameter -workDir
+# * JENKINS_USERNAME : username to use when authenticating to master -credentials
+# * JENKINS_PASSWORD : password to use when authenticating to master -credentials
+
+
+# Get the secret from Jenkins master
+get_secret() {
+	local jenkins_url="$1"
+	local jenkins_agent_name="$2"
+	local username="$3"
+	local password="$4"
+	local crumb=".crumb:dac7ce5614f8cb32a6ce75153aaf2398"
```

Comment:
This CRUMB should be retrieved dynamically for instances

Reply:
Updated to get the crumb dynamically. Thanks for the tip @oleg-nenashev.

Note: Interestingly, just tested several times and it works just fine even with hardcoded crumb despite CSRF is properly configured on master.